client/fingerprint/cpu.go

```diff
@@ -64,12 +72,13 @@ func (f *CPUFingerprint) Fingerprint(req *FingerprintRequest, resp *FingerprintR
 		tt = cfg.CpuCompute
 	}
 
-	// Return an error if no cpu was detected or explicitly set as this
-	// node would be unable to receive any allocations.
+	// If we cannot detect the cpu total compute, fallback to a very low default
+	// value and log a message about configuring cpu_total_compute. This happens
+	// on Graviton instances where CPU information is unavailable. In that case,
+	// the env_aws fingerprinter updates the value with correct information.
 	if tt == 0 {
-		return fmt.Errorf("cannot detect cpu total compute. "+
-			"CPU compute must be set manually using the client config option %q",
-			"cpu_total_compute")
+		f.logger.Info("fallback to default cpu total compute, set client config option cpu_total_compute to override")
```

Comment:
The log line is a little unfortunate in that it's important to see in non-AWS cases but in AWS cases we're going to fix it in the fingerprinter, which logs at debug. But I don't see a good way around that other than dropping this one to debug... which I think would be a bad idea?

Reply:
Yeah I could be convinced to pick any of the possible combinations of Info/Debug. Realistically the most correct solution to all of this would involve plumbing to enable fingerprinters to coordinate. But for this, that would be like driving a thumbtack with sledgehammer.